lib/cloudflare_challenge.rb

```diff
@@ -6,11 +6,14 @@ class CloudflareChallenge
   class NoCloudflareAPIKey < StandardError; end;
   class NoCloudflareEmail < StandardError; end;
 
-  attr_reader :zone, :domains, :client, :challenges, :token
+  attr_reader :zone, :domains, :client, :challenges, :token, :email, :api_key
 
   def initialize(options = {})
-    raise NoCloudflareAPIKey if ENV['CLOUDFLARE_API_KEY'].blank?
-    raise NoCloudflareEmail  if ENV['CLOUDFLARE_EMAIL'].blank?
+    @email   = options[:email] || ENV['CLOUDFLARE_EMAIL']
```

Comment:
Consider using no fallback to ENV variables. Supporting 2 methods may be too cumbersome